### DISCUSSION TITLE
JuiceFS S3 Gateway development plan

### DISCUSSION THREAD

**Discussion by CalmVibes** [juicedata/juicefs#2351] [General]
First of all, I want to thank you for a wonderful product that helped solve some of our tasks. I want to take an interest about the future of Juicefs S3 Gateway and ask a few questions. We use JuiceFS S3 Gateway in Kubernetes, so we have no way to compile it from source.

- Is there a development plan, roadmap?
- Is there a history of releases and changes in versions?
- Is it possible to create a public bucket?
- Is it possible to separate endpoints for HTTPS and S3 API?
- Are there any plans to develop Juicefs S3 Gateway UI and functionality to the level of of MinIO gateways such as multi-user management, multi buckets, public buckets, etc.?

Thank you.

**Comment by zhijian-pro** [comment]
We have implemented juicefs as the backend of minio s3gateway. You can see this [repository](https://github.com/juicedata/minio/tree/gateway). 
It has the full function of minio s3gateway. Because the minio s3gateway function is relatively perfect, so we don't have a new development plan for the time being.

**Reply by CalmVibes** [reply]
Thank you for your reply!

Can we use this [repository](https://github.com/juicedata/minio/tree/gateway) in Kubernetes deployment?

We used this [instruction](https://github.com/juicedata/juicefs/blob/main/docs/en/deployment/s3_gateway.md) and got this Console without full functionality of MinIO gateway such as multi-user management: 
![image](https://user-images.githubusercontent.com/64099513/178451990-c7f06532-fa5c-48cf-9000-074ef98053d9.png)

**Reply by xiaogaozi** [reply]
You need to build a Docker image containing the fully functional MinIO Gateway binary by yourself, and refer to [this YAML](https://github.com/juicedata/juicefs/blob/main/deploy/juicefs-s3-gateway.yaml) to deploy to K8s.

**Reply by CalmVibes** [reply]
@xiaogaozi Thank you for your answer!

**Comment by zhijian-pro** [comment]
@CalmVibes  You may wish to be notified of this information https://github.com/juicedata/juicefs/pull/4030
